samples/python/onedim/diffusion_flame_continuation.py

```diff
@@ -54,43 +62,139 @@
+            temperature_increment *= 0.9 * target_delta_Tmax / (abs(max(f.T) - Tmax))
+        error_count = 0
+    except ct.CanteraError as err:
+        logger.debug(err)
+        if strain_rate / a_max < strain_rate_tol:
+            logger.info('Traversed unstable branch down to '
+                f'{100 * strain_rate / a_max:.1f}% of the maximum strain rate.')
+            break
+
+        # Restore the previous solution and try a smaller temperature increment for the
+        # next iteration
+        f.from_array(backup_state)
+        temperature_increment = 0.7 * temperature_increment
+        error_count += 1
+        logger.warning(f"Solver did not converge on iteration {i}. Trying again with "
+                       f"dT = {temperature_increment:.2f}")
+
+    if ct.hdf_support():
+        f.save(output_path / 'flame_profiles.h5', name=f'iteration{i}', overwrite=True)
+
+    # Collect output stats
+    Tmax = max(f.T)
+    Tmid = 0.5 * (min(f.T) + max(f.T))
+    s = np.where(f.T > Tmid)[0][[0,-1]]
+    width = f.grid[s[1]] - f.grid[s[0]]
+    strain_rate = f.strain_rate('max')
+    a_max = max(strain_rate, a_max)
+
+    data.append({
+        'Tmax': max(f.T),
+        'strain_rate': strain_rate,
+        'heat_release_rate': np.trapz(f.heat_release_rate, f.grid),
+        'n_points': len(f.grid),
+        'flame_width': width,
+        'Tc_increment': temperature_increment,
+        'time_steps': sum(f.time_step_stats),
+        'eval_count': sum(f.eval_count_stats),
+        'cpu_time': sum(f.jacobian_time_stats + f.eval_time_stats),
+        'errors': error_count
+    })
+
+    if error_count >= max_error_count:
+        logger.warning(f'Stopping after {error_count} successive solver failures.')
         break
 
-    maximum_temperature.append(np.max(f.T))
-    a_max.append(np.max(np.abs(np.gradient(f.velocity) / np.gradient(f.grid))))
+logger.info(f'Stopped after {i} iterations')
 
+# %%
+# Combine data
+# ------------
+df = pd.DataFrame.from_records(data)
+df.to_csv(output_path / f'integral_data.csv')
+df
 
+# %%
 # Plot the maximum temperature versus the maximum axial velocity gradient
+# -----------------------------------------------------------------------
 plt.figure()
-plt.plot(a_max, maximum_temperature)
+plt.plot(df.strain_rate, df.Tmax)
 plt.xlabel('Maximum Axial Velocity Gradient [1/s]')
 plt.ylabel('Maximum Temperature [K]')
 plt.savefig(output_path / "figure_max_temperature_vs_max_velocity_gradient.png")
 
-
+# %%
 # Plot maximum_temperature against number of iterations
+# -----------------------------------------------------
 plt.figure()
-plt.plot(range(len(maximum_temperature)), maximum_temperature)
+plt.plot(df.Tmax)
 plt.xlabel('Number of Continuation Steps')
 plt.ylabel('Maximum Temperature [K]')
-plt.savefig(output_path / "figure_max_temperature_iterations.png")
\ No newline at end of file
+plt.savefig(output_path / "figure_max_temperature_iterations.png")
```

Comment:
Here and on the corresponding line above. You need to include a 
```python
plt.show()
```
so the plot is actually shown when running the script from the command line..